### PR TITLE
Enable the Recovery Phrase Verification feature on ITN

### PR DESCRIPTION
**Pull request by daniloprates**
This PR enables the Recovery Phrase Verification feature of Shelley wallets on ITN

## Todos

- [x] Enable the Recovery Phrase Verification on ITN
- [x] Update test coverage

## Screenshots

Screen Shot 2020-05-12 at 15 48 26
Screen Shot 2020-05-12 at 15 48 37
Screen Shot 2020-05-12 at 15 49 27
Screen Shot 2020-05-12 at 15 49 33

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/components/wallet/settings/WalletRecoveryPhraseStep2Dialog.js [hunk 1]
```diff
@@ -76,6 +76,10 @@ export default class WalletRecoveryPhraseStep2Dialog extends Component<
     intl: intlShape.isRequired,
   };
 
+  static defaultProps = {
+    wordCount: 15,
```
Comment (nikolaglumac) [inline source/renderer/app/components/wallet/settings/WalletRecoveryPhraseStep2Dialog.js:80]: @daniloprates please replace hardcoded number with constant: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/config/cryptoConfig.js#L4

File: source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js [hunk 2]
```diff
@@ -131,6 +129,10 @@ export default class WalletRecoveryPhraseVerificationWidget extends Component<Pr
     intl: intlShape.isRequired,
   };
 
+  static defaultProps = {
+    wordCount: 15,
+  };
```
Comment (nikolaglumac) [inline source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js:134]: @daniloprates please replace hardcoded number with constant: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/config/cryptoConfig.js#L4

**Comment by nikolaglumac** [comment]
@gnpf @ManusMcCole @tomothespian can you please review this one?

**Comment by nikolaglumac** [comment]
@daniloprates we also have failing tests... The Byron builds no longer show the verification feature at all :(

Screenshot 2020-05-13 at 10 45 20
recovery-phrase-verification-2020-05-13T084514 0545Z

**Comment by nikolaglumac** [comment]
Byron build:

**Daedalus wallet**
Screenshot 2020-05-13 at 10 47 27


**Yoroi wallet**
Screenshot 2020-05-13 at 10 47 34

**Comment by nikolaglumac** [comment]
When running test on the ITN selfnode I get this failure:

Screenshot 2020-05-13 at 10 57 24
Screenshot 2020-05-13 at 10 57 09

**Comment by daniloprates** [comment]
@nikolaglumac fixed all the issues mentioned.

File: source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js [hunk 3]
```diff
@@ -235,16 +233,7 @@ export default class WalletRecoveryPhraseVerificationWidget extends Component<Pr
       <div className={styles.component}>
         <h2>{intl.formatMessage(messages.title)}</h2>
         <div className={styles.description}>
-          {intl.formatMessage(messages.description, {
```
Comment (nikolaglumac) [inline source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js:238]: @daniloprates why did we remove this "paper wallet" tooltip part?

Comment (daniloprates) [reply source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js:238]: Missed that when re-enabling the Byron bit. Done, now.

File: source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js [hunk 4]
```diff
@@ -235,16 +254,19 @@ export default class WalletRecoveryPhraseVerificationWidget extends Component<Pr
...
-          })}{' '}
-          <Tooltip
-            className={styles.paperWallet}
-            skin={TooltipSkin}
-            tip={intl.formatMessage(messages.paperWalletDescription)}
-          >
-            {intl.formatMessage(messages.paperWalletTitle)}
-          </Tooltip>
+          {intl.formatMessage(messages.description, { wordCount })}
+          {!isIncentivizedTestnet && (
```
Comment (nikolaglumac) [inline source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js:258]: @daniloprates it is possible to restore a paper wallet on the ITN too. Therefore, I believe we don't need this switch...

Comment (daniloprates) [reply source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js:258]: hey @nikolaglumac, for ITN we only display this widget for Shelley wallets, that's why we need the conditional

Comment (nikolaglumac) [reply source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js:258]: oh I forgot that 🤦 sorry!

**Pull request merged**